src/client/renderer/r_state.c

```diff
@@ -391,41 +358,99 @@ void R_UploadToSubBuffer(r_buffer_t *buffer, const size_t start, const size_t si
 		// it's fairly rare you'll be editing at the end first,
 		// but who knows.
 		if (start) {
+
 			glBufferData(buffer->target, total_size, NULL, buffer->hint);
 			R_GetError("Partial resize");
+			r_view.num_buffer_full_uploads++;
 			glBufferSubData(buffer->target, start, size, data);
 			R_GetError("Partial update");
+			r_view.num_buffer_partial_uploads++;
 		} else {
 			glBufferData(buffer->target, total_size, data, buffer->hint);
 			R_GetError("Full resize");
+			r_view.num_buffer_full_uploads++;
 		}
 
+		r_state.buffers_total_bytes -= buffer->size;
+		r_state.buffers_total_bytes += total_size;
+
 		buffer->size = total_size;
 	} else {
 		// just update the range we specified
 		glBufferSubData(buffer->target, start, size, data);
-
 		R_GetError("Updating existing buffer");
+		r_view.num_buffer_partial_uploads++;
 	}
 
-	r_view.num_buffer_uploads++;
 	r_view.size_buffer_uploads += size;
 }
 
 /**
+ * @brief
+ */
+static GLubyte R_GetElementSize(const GLenum type) {
+
+	switch (type) {
+		case R_ATTRIB_BYTE:
+		case R_ATTRIB_UNSIGNED_BYTE:
+		case R_ATTRIB_INT_2_10_10_10_REV:
+			return 1;
+		case R_ATTRIB_SHORT:
+		case R_ATTRIB_UNSIGNED_SHORT:
+			return 2;
+		case R_ATTRIB_INT:
+		case R_ATTRIB_UNSIGNED_INT:
+		case R_ATTRIB_FLOAT:
+			return 4;
+		default:
+			Com_Error(ERR_DROP, "Bad GL type");
+	}
+}
+
+/**
+ * @brief
+ */
+uint32_t R_GetNumAllocatedBuffers(void) {
+	return r_state.buffers_total;
+}
+
+/**
+ * @brief
+ */
+uint32_t R_GetNumAllocatedBufferBytes(void) {
+	return r_state.buffers_total_bytes;
+}
+
+/**
  * @brief Allocate a GPU buffer of the specified size.
  * Optionally upload the data immediately too.
  */
-void R_CreateBuffer(r_buffer_t *buffer, const GLenum hint, const r_buffer_type_t type, const size_t size,
+void R_CreateBuffer(r_buffer_t *buffer, const r_attrib_type_t element_type, const GLubyte element_count,
+                    const _Bool element_normalized, const GLenum hint,
+                    const r_buffer_type_t type, const size_t size,
                     const void *data) {
 
-	assert(buffer->bufnum == 0);
+	if (buffer->bufnum != 0) {
+		Com_Warn("Attempting to reclaim non-empty buffer");
```

Comment:
This one, too, is not something users should typically see. Making it a Com_Debug makes sense imo.